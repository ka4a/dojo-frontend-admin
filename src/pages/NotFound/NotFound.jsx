import React, { memo } from 'react';
import { Hyperlink } from '@edx/paragon';
import { getConfig } from '@edx/frontend-platform';
import { compose } from 'redux';
import {
  FormattedMessage,
  injectIntl,
} from '@edx/frontend-platform/i18n';
import Header from '@reustleco/dojo-frontend-component-admin-header';

const wrapper = compose(
  memo,
  injectIntl,
);

export const NotFound = wrapper(({ intl }) => {
  const homepageMessage = {
    id: 'error.notfound.homepageMessage',
    defaultMessage: 'Homepage',
    description: 'Link to userâ€™s Homepage',
  };

  const dashboardLink = (
    <Hyperlink
      variant="muted"
      isInline
      destination={`${getConfig().LMS_BASE_URL}/dashboard`}
    >
      {intl.formatMessage(homepageMessage)}
    </Hyperlink>
  );

  return (
    <div>
      <Header />
      <div className="container-fluid justify-content-center align-items-center text-center">
        <h1 className="text-muted">
          <FormattedMessage
            id="error.notfound.header"
            defaultMessage="Page Not Found"
            description="error message when a page does not exist"
          />
        </h1>
        <p className="text-muted">
          <FormattedMessage
            id="error.notfound.message"
            defaultMessage="The page that you were looking for was not found. Go back to the {dashboardLink} ."
            description="error message when a page does not exist"
            values={{ dashboardLink }}
          />
        </p>
      </div>
    </div>
  );
});
