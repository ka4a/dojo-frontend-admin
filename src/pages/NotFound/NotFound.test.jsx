import { IntlProvider } from '@edx/frontend-platform/i18n';
import { shallow, mount } from 'enzyme';
import { NotFound } from './NotFound';

describe('NotFound', () => {
  it('should render NotFound component', () => {
    shallow(
      <IntlProvider locale="en">
        <NotFound />
      </IntlProvider>,
    );
  });

  it('it should render the expected HTML', () => {
    expect(
      mount(
        <IntlProvider locale="en">
          <NotFound />
        </IntlProvider>,
      ).html(),
    ).toMatchSnapshot();
  });
});
