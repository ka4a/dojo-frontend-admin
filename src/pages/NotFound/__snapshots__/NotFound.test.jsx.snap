// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NotFound it should render the expected HTML 1`] = `"<div><div class=\\"container-fluid justify-content-center align-items-center text-center\\"><h1 class=\\"text-muted\\"><span>Page Not Found</span></h1><p class=\\"text-muted\\"><span>The page that you were looking for was not found. Go back to the <a class=\\"pgn__hyperlink muted-link inline-link\\" href=\\"http://localhost:18000/dashboard\\" target=\\"_self\\">Homepage</a> .</span></p></div></div>"`;
